Allow setting logger output format with TORCH_LOGS_FORMAT

oulgen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #111808
* __->__ #111770

TORCH_LOGS_FORMAT="%(levelname)s: %(message)s" will only dump output
level and message contents

oulgen committed:
Allow setting logger output format with TORCH_LOGS_FORMAT

TORCH_LOGS_FORMAT="%(levelname)s: %(message)s" will only dump output
level and message contents

[ghstack-poisoned]

oulgen committed:
Update on "Allow setting logger output format with TORCH_LOGS_FORMAT"


TORCH_LOGS_FORMAT="%(levelname)s: %(message)s" will only dump output
level and message contents

[ghstack-poisoned]

voznesenskym commented on torch/_logging/_internal.py:
```diff
@@ -663,7 +667,15 @@ def format(self, record):
+    if fmt is None:
+        return TorchLogsFormatter()
+    else:
+        return logging.Formatter(fmt)
```
What if the formatter is invalid, what kind of error do we get?

oulgen replied:
Currently, we throw a `ValueError: Formatting field not found in record` error. We could find a validate it but not sure how valuable/easy it would be.

voznesenskym commented on torch/_logging/_internal.py:
```diff
@@ -515,6 +516,9 @@ def pad_to(s, length=30):
   TORCH_LOGS="+some.random.module,schedule" will set the log level of
   some.random.module to logging.DEBUG and enable the schedule artifact
+
+  TORCH_LOGS_FORMAT="%(levelname)s: %(message)s" or any provided format
```
what are all the valid keys here? What about frame counts? timestamps?

oulgen replied:
I can list out all the valid keys

voznesenskym commented:
I get the desire to want to make it general purpose, but it really just seems like you want all annotations, or none (just the message).

oulgen commented:
@voznesenskym This is customizable enough to appease all different opinions, which is why i went with this version. There was not a consensus around simpler versions. Since this is not a publicly announced API, it should be okay to be less precise/hardened around it. What do you think?

Chillee commented:
I wonder whether we should turn the default to not have that info

oulgen commented:
@Chillee I proposed that but @voznesenskym was against it. It will probably be hard to get consensus on something like this.

oulgen committed:
Update on "Allow setting logger output format with TORCH_LOGS_FORMAT"


TORCH_LOGS_FORMAT="%(levelname)s: %(message)s" will only dump output
level and message contents

[ghstack-poisoned]

oulgen commented:
@pytorchbot merge
